app/src/main/java/org/sagebase/crf/step/active/BpmRecorder.java

```diff
@@ -224,8 +232,27 @@ public HeartBeatJsonWriter(BpmUpdateListener
         @Override
         public void onHeartRateSampleDetected(HeartBeatSample sample) {
             bpmCalculator.calculateBpm(sample);
-            
-            mJsonObject.addProperty(TIMESTAMP_IN_SECONDS_KEY,  sample.t / 1_000);
+
+            if (timestampZeroReference < 0) {
+                // set timestamp reference, which timestamps are measured relative to
+                timestampZeroReference = sample.t;
+                uptimeZeroReference = System.nanoTime() * 1e-9;
+
+                Date timestampReferenceDate = new Date(System.currentTimeMillis());
+                // TODO: syoung 10/04/2018 Figure out how to set timestamp dates to "en_US_POSIX"
+                mJsonObject.addProperty(TIMESTAMP_DATE_KEY,
+                        new SimpleDateFormat(FormatHelper.DATE_FORMAT_ISO_8601, new Locale("en", "us", "POSIX"))
+                                .format(timestampReferenceDate));
```

Comment:
@nategbrown9  Could you pleasea remove Shannon's TODO above?

We also should make this change in SensorRecorder, LocationRecorder, ConsentSignatureStepLayout, and FormatHelper.